plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/coupon/style.scss

```diff
@@ -13,23 +13,23 @@
 	display: flex;
 	width: 100%;
 	margin-bottom: 0;
+	gap: $gap-smaller;
```

Comment:
Using `gap` instead of margins so that spacing is maintained between elements when fields wrap.